test/src/main/java/org/apache/accumulo/test/functional/HalfDeadTServerIT.java

```diff
@@ -63,7 +66,29 @@ protected int defaultTimeoutSeconds() {
     return 4 * 60;
   }
 
-  class DumpOutput extends Daemon {
+  private static final AtomicBoolean sharedLibBuilt = new AtomicBoolean(false);
+
+  @SuppressFBWarnings(value = "COMMAND_INJECTION",
+      justification = "command executed is not from user input")
+  @BeforeClass
+  public static void buildSharedLib() throws IOException, InterruptedException {
+    String root = System.getProperty("user.dir");
+    String source = root + "/src/test/c/fake_disk_failure.c";
+    String lib = root + "/target/fake_disk_failure.so";
```

Comment:
If the so exists, do you even need to rec-compile? I guess you run the risk of it being corrupt or an incorrect so, but a clean should remove this as well.

Reply:
Perhaps not, but that's a bit out of scope of what I was trying to optimize here. I was actually thinking about creating a proper Makefile and building the shared lib with exec-maven-plugin rather than doing it with a direct call to gcc in the test. But again, I felt it was a bit out of scope.